src/commands/CreateVDBCommand.ts

```diff
@@ -16,38 +16,24 @@
  */
 import * as constants from '../constants';
 import * as extension from '../extension';
-import * as fs from 'fs';
 import * as path from 'path';
 import * as utils from '../utils';
 import * as vscode from 'vscode';
 import { DataVirtConfig } from '../model/DataVirtModel';
 
-export function createVDBCommand() {
+export async function createVDBCommand() {
 	if (extension.workspaceReady) {
-		vscode.window.showInputBox( { validateInput: (name: string) => {
-			let res: string = utils.validateName(name);
-			if (!res) {
-				res = utils.validateFileNotExisting(name);
-			}
-			return res;
-		}, placeHolder: 'Enter the name of the new virtual database' })
-			.then( (vdbName: string) => {
-				if (vdbName === undefined) {
-					return;
-				}
-				handleVDBCreation(vscode.workspace.workspaceFolders[0].uri.fsPath, vdbName)
-					.then( (success: boolean) => {
-						if (success) {
-							utils.openDDLEditor(vdbName);
-							vscode.window.showInformationMessage(`Virtual database ${vdbName} has been created successfully...`);
-						} else {
-							vscode.window.showErrorMessage(`An error occured when trying to create a new virtual database with the name ${vdbName}...`);
-						}
-					})
-					.catch( (error) => {
-						extension.log(error);
-					});
-			});
+		const vdbName: string = await vscode.window.showInputBox( { validateInput: utils.validateVDBName, placeHolder: 'Enter the name of the new virtual database' });
+		if (vdbName === undefined) {
+			return;
+		}
+		const success: boolean = await handleVDBCreation(vscode.workspace.workspaceFolders[0].uri.fsPath, vdbName);
```

Comment:
Is there a reason we're using the root workspace folder and not going with whatever folder the user has selected in the Explorer?

Reply:
Tbh that part is not yet specified very clearly atm. We are only processing YAML files in the root folder for now until we define it differently. (and that will end up in a separate Jira anyways)

Reply:
Ok. There's code in Camel K that you can use to get the current folder I think, so when you get there give me a holler.

Reply:
for another PR: worth mentioning in readme that the initial version is supporting only a single folder in VS Code workspace

Reply:
I asked Melissa to include that in the documentation.